xamarin/Xamarin.Forms
# [MacOS] Fix resetting TextColor and Font on Label

### Description of Change ###

Reorder setting the initial properties for `TextColor` and `Font` on MacOS & iOS to overcome losing values in subsequent calls to set other properties.

#### Background 

Whenever you set `StringValue` it will automatically fill `AttributedStringValue` as well. On iOS (while property names are slightly different) it seems to be no problem if you manipulate the attributed string directly, for instance when you set the character spacing and then set the text color. It will update it for you accordingly.

However, on MacOS that doesn't seem to be the case. From what I can see, whenever you change something in the `AttributedStringValue` it will lose sync and then trying to set the font or text color will have no effect. Unfortunately, unlike for iOS, the behavior in this area is not documented for macOS.

I also removed code that was initially dropped in to overcome sort of the same problem, but effectively was duplicating functionality in a different place.

### Issues Resolved ### 

- fixes #10101

### API Changes ###

 None

### Platforms Affected ### 


- iOS
- MacOS

### Behavioral/Visual Changes ###
Should be none, other than that everything starts working as it should 🎉 

### Before/After Screenshots ### 


Not applicable

### Testing Procedure ###
Navigate to issue 10101 in the control gallery and observe how properties are being preserved when we set a different text.

In addition, make sure that all styling on a label still works as it should on both iOS and MacOS.

### PR Checklist ###
